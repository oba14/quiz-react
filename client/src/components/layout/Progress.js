import React from "react";

function Progress({ current, total }) {
  return (
    <h2 data-testid="progress-check">
      Question {current} of {total}
    </h2>
  );
}

export default Progress;
