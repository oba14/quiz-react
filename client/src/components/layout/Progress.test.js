import React from "react";
import ReactDOM from "react-dom";
import Progress from "./Progress";
import { render, screen, fireEvent, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

afterEach(cleanup);

describe("Test Progress Component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Progress></Progress>, div);
  });

  it("renders progrees correctly", () => {
    const { getByTestId } = render(
      <Progress current={4} total={25}></Progress>
    );
    expect(getByTestId("progress-check")).toHaveTextContent("Question 4 of 25");
  });

  it("matched snapshot", () => {
    const tree = renderer
      .create(<Progress current={6} total={30}></Progress>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
