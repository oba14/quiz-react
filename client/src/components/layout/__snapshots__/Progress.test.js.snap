// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matched snapshot 1`] = `
<h2
  data-testid="progress-check"
>
  Question 
  6
   of 
  30
</h2>
`;
